File: Competitive Programming/DAY-21/angle.py
def anglecalc(hours,minutes):
    h=hours
    m=minutes
    if(h<0 or m<0 or h>60 or m>60):
        print("Give correct input")
    elif h==12:
        h=0
    elif m==60:
        m=0
        h=h+1
        if(h>12):
            h=h-12
    h_angle=(h*60+m)*0.5
    m_angle=(m*6)
    angle=abs(h_angle-m_angle)
    angle=min(360-angle,angle)
    return int(angle)

ho=int(input("Enter hour"))
mi=int(input("Enter minute"))
anglehm=anglecalc(ho,mi)
if(anglehm==0):
    print("Hour and minute overlapped")
else:
    print(f"The angle is {anglehm} degrees")

File: Competitive Programming/DAY-21/angle.cpp
#include <iostream>
#include <math.h>
using namespace std;

int min(int x, int y) 
{ 
    return (x < y)? x: y; 
     
}

int anglecalc(int hours,int minutes)
{
    if(hours<0||minutes<0||hours>12||minutes>60)
    cout<<"Wrong Input";
    else if(hours==12)
    hours=0;
    else if(minutes==60)
    {
        minutes=0;
        hours+=1;
        if(hours>12)
        {
            hours=hours-12;
        }
    }
    float hour_angle=(hours*60+minutes)*0.5;
    float minute_angle=6*minutes;
    
    float angle=abs(hour_angle-minute_angle);
    
    angle=min(360-angle,angle);
    
    return angle;
    
}

int main()
{
    int hours,minutes;
    cout<<"Enter hours";
    cin>>hours;
    cout<<"Enter minutes";
    cin>>minutes;
    int c=anglecalc(hours,minutes);   
    if(c==0)
    cout<<"Hour and minute overlapped";
    else
    cout<<"Angle is"<<c<<"degrees";
    return 0;
}
